app/src/main/java/io/homeassistant/companion/android/webview/WebViewActivity.kt

```diff
@@ -426,6 +444,13 @@ class WebViewActivity : AppCompatActivity(), io.homeassistant.companion.android.
             webView.evaluateJavascript("externalBus(${JSONObject(message)})") {
                 Log.d(TAG, "NFC Write Complete $it")
             }
+        } else if (requestCode == FILE_CHOOSER_RESULT_CODE) {
+            val results: Uri? = data?.data
+            if (results != null)
+                mFilePathCallback!!.onReceiveValue(arrayOf(results))
+            else
+                mFilePathCallback?.onReceiveValue(arrayOf())
+            mFilePathCallback = null
```

Comment:
```suggestion
            mFilePathCallback?.onReceiveValue(WebChromeClient.FileChooserParams.parseResult(resultCode, intent))
            mFilePathCallback = null
```

Reply:
Had to replace `intent` with `data` as data is what we get in this method, otherwise we never send the image.